driver-core/src/main/com/mongodb/connection/CommandMessage.java

```diff
@@ -72,7 +75,8 @@ boolean containsPayload() {
     }
 
     boolean isResponseExpected() {
-        return !useOpMsg() || responseExpected;
+        isTrue("The message must be encoded before determining if a response is expected", messageHasBeenEncoded);
```

Comment:
If you follow by suggestions above, this isTrue() check will cause a problem if you start calling this method from getFlagBits() but you can work around that by creating a private method that computes the value.

Reply:
Done.